### PR TITLE
Data migration improvements

### PR DESCRIPTION
**Pull request by buckysballs**

### CODE REVIEW

File: src/main/scala/org/constellation/rollback/RollbackAccountBalances.scala [hunk 1]
```diff
@@ -27,6 +27,7 @@ class RollbackAccountBalances {
     if (snapshotHash == zeroSnapshotHash) {
       balances
     } else {
+      println(snapshotHash)
```
Comment (mchrapek) [inline src/main/scala/org/constellation/rollback/RollbackAccountBalances.scala]: should be removed

File: src/main/scala/org/constellation/consensus/EdgeProcessor.scala [hunk 2]
```diff
@@ -231,11 +241,106 @@ case class SnapshotInfo(
...
+
+  def writeLocal(infoSer: SnapshotInfoSer = this, basePath: String = "rollback_data/snapshot_info/") =
+    write[Unit](localWriterProc)(infoSer, basePath)
+}
 
 case object GetMemPool
 
-case class Snapshot(lastSnapshot: String, checkpointBlocks: Seq[String]) extends Signable
+case class Snapshot(lastSnapshot: String, checkpointBlocks: Seq[String]) extends Signable {
+//  val snapshotHash = this.hash
```
Comment (mchrapek) [inline src/main/scala/org/constellation/consensus/EdgeProcessor.scala]: don't we need it?

File: src/test/scala/org/constellation/rollback/RollbackLoaderTest.scala [hunk 3]
```diff
@@ -10,12 +10,12 @@ class RollbackLoaderTest extends FreeSpec with ArgumentMatchersSugar with Before
...
   private val loaderForExistingFolder = new RollbackLoader(
     existingFolder + "snapshots",
-    existingFolder + "rollback_info",
+    existingFolder + "snapshot_info",
     existingFolder + "rollback_genesis"
   )
   private val loaderForNotExistingFolder = new RollbackLoader(
     notExistingFolder + "snapshots",
-    notExistingFolder + "rollback_info",
+    notExistingFolder + "snapshot_info",
```
Comment (mchrapek) [inline src/test/scala/org/constellation/rollback/RollbackLoaderTest.scala]: `NotExistingFolder` and `ExistingFoler` are the same

File: src/main/scala/org/constellation/consensus/EdgeProcessor.scala [hunk 4]
```diff
@@ -217,6 +217,16 @@ object EdgeProcessor extends StrictLogging {
       "handleSignatureRequest"
     )(ConstellationExecutionContext.bounded, dao)
 
+  def chunkSerialize[T](chunk: Seq[T], tag: String) = {
+    logger.warn(s"chunkSerialize ${tag}")
+    KryoSerializer.serializeAnyRef(chunk)
+  }
+
+  def chunkDeSerialize[T](chunk: Array[Byte], tag: String) = {
+    logger.warn(s"chunkDeSerialize ${tag}")
```
Comment (mchrapek) [inline src/main/scala/org/constellation/consensus/EdgeProcessor.scala]: I think it shouldn't be warn level

File: src/main/scala/org/constellation/storage/SnapshotService.scala [hunk 5]
```diff
@@ -95,27 +96,38 @@ class SnapshotService[F[_]: Concurrent](
...
-            .use(
-              stream =>
-                Sync[F].delay {
-                  stream.write(KryoSerializer.serializeAnyRef(info))
-                }.flatTap { _ =>
-                  logger.debug(s"SnapshotInfo written for hash: ${info.snapshot.hash} in path: ${path}")
-                }
-            )
+          for {
+            _ <- writeSnapshotInfoParts(info, overWritePath)
```
Comment (mchrapek) [inline src/main/scala/org/constellation/storage/SnapshotService.scala]: We can write more simply `writeSnapshotInfoParts(info, overWritePath).map(_ => ())`

**Comment by mchrapek** [comment]
I will take this branch and I will continue working here

**Comment by marcinwadon** [comment]
Closing the PR in favor of #762 which is continuation of this PR

**Pull request closed without merging**